lib/core/src/Cardano/Wallet/Primitive/Types/Tx.hs

```diff
@@ -953,6 +956,15 @@ mockSealedTx :: HasCallStack => ByteString -> SealedTx
 mockSealedTx = SealedTx False
     (internalError "mockSealedTx: attempted to decode gibberish")
 
+{-------------------------------------------------------------------------------
+                          Checks
+-------------------------------------------------------------------------------}
+
+coinIsValidForTxOut :: Coin -> Bool
+coinIsValidForTxOut c = (&&)
+    (c >= txOutMinCoin)
+    (c <= txOutMaxCoin)
```

Comment:
I agree that we may want to avoid hard-coding the minimum and maximum value for `Coin` — they are actually configured in the Shelley genesis. (In fact, our local cluster has a higher maximum Ada quantity than 45B.)

In fact, I believe that we can remove `txOutMaxCoin` entirely — checking the maximum bound will never catch any user mistakes that will not manifest in a different way (e.g. inability to find inputs with sufficient funds).

When discussing this, Jonathan suggested to make a second PR in order to keep the scope of this PR to a refactoring (as opposed to a change in functionality).

Reply:
@Anviking I completely agree with your observation. This value could be different in custom testnets, so hard-coding it within the wallet primitive layer is of questionable value.

Though as @HeinrichApfelmus mentioned, it was my aim to avoid changing any behaviour in this PR.

I think we could/should consider making a ticket to capture this issue. I'll try to do that today.